report/report.md

```diff
@@ -0,0 +1,898 @@
+heavily relies on, the authors presented a generalized version of the
+previous accumulation structure of Halo that they coined _Accumulation
+Schemes_. Simply put, given a predicate $\Phi: X \to \Bb$,
+an accumulation scheme consists of the following functions:
+
+- $\ASProver(q_i: X, acc_i: \Acc) \to \Acc$
+
+    The prover accumulates the instance $q_i$ into the previous accumulator
+    $acc_{i-1}$ into the new accumulator $acc_i$.
+
+- $\ASVerifier(q_i: X, acc_i: \Acc, acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+    The verifier checks that the instance $q_i$ was correctly accumulated
+    into the previous accumulator $acc_{i-1}$ to form the new accumulator
+    $acc_i$.
+
+- $\ASDecider(acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+    The decider performs a single check that simultaneously ensures that
+    _every_ previous instance-proof pair accumulated in $acc_{i+1}$ verifies
+    assuming the $\ASVerifier$ has accepted that every previous accumulator
+    correctly accumulates $q_i$.
+
+We define completeness and soundness for the Accumulation Scheme:
+
+- **Completeness:** For all accumulators $acc_i$ and predicate inputs $q \in X$,
+  if $\top = \ASDecider(acc) = \Phi(q)$, then for $\ASProver(q, acc_i)
+  = acc_{i+1}$ it holds that $\top = \ASVerifier(acc_i, q, acc_{i+1}) =
+  \ASDecider(acc_{i+1})$.
+
+- **Soundness:** For all efficiently-generated accumulators $acc_i, acc_{i+1}
+  \in \Acc$ and predicate inputs $q \in X$, if $\top = \ASDecider(acc_{i+1}) =
+  \ASVerifier(q, acc_i, acc_{i+1})$ then, with all but negligible probability,
+  $\top = \Phi(q) = \ASDecider(acc_i)$.
+
+### IVC from Accumulation Schemes
+
+If the $\ASVerifier$ runtime is sub-linear and we have a $\SNARKVerifierSlow$
+that may run in linear time, then we can create an IVC scheme:
+
+- $\IVCProver(z_i: \textbf{Instance}, \pi_i: \textbf{Proof}, acc_i: \Acc) \to (\textbf{Proof}, \Acc)$
+
+  - Accumulates $(z_i, \pi_i)$ with $acc_i$ to obtain a new accumulator $acc_{i+1}$.
+  - Then generates a SNARK proof $\pi_{i+1}$ of the following claim, expressed as a circuit $R$:
+      $$"z_{i+1} = F(z_i) \; \land \; (i = 0 \lor \exists \, \pi_i, \, acc_i \text{ s.t. } \ASVerifier((z_i, \pi_i), acc_i, acc_{i+1}) = \top)"$$
+  - Output $(\pi_{i+1}, acc_{i+1})$
+
+- $\IVCVerifier(\pi_{i+1}: \textbf{Proof}, acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+  Checks the proof: $\top \meq \SNARKVerifierSlow(\pi_{i+1}) \meq \ASDecider(acc_{i+1})$
+
+Consider the above chain run $n$ times. As in the "simple" SNARK IVC
+construction, if the SNARK verifier accepts at the end, then that means:
+
+$$
+\begin{alignedat}[b]{2}
+  &\SNARKVerifierSlow(\pi_i)                               &&= \top \then \\
+  &\ASVerifier((z_{n-1}, \pi_{n-1}), acc_{n-1}, acc_n)     &&= \top \then \\
+  &\ASVerifier((z_{n-2}, \pi_{n-2}), acc_{n-2}, acc_{n-1}) &&= \top \then \cdots \\
+  &\ASVerifier((z_0, \pi_0), acc_0, acc_1)                 &&= \top \then \\
+\end{alignedat}
+$$
```

Comment:
Also just putting a pin on this, I need to recurse to z_1 instead as the verifier does not run on z_0, in my description.